### PR TITLE
modern C++ #2 - homework

### PR DESCRIPTION
**Pull request by niemarcin**

### CODE REVIEW

**Review by MaciejSalamonski — APPROVED** [review]
I dont have any comments :).

File: shapes/main.cpp [hunk 1]
```diff
@@ -92,9 +85,19 @@ int main()
...
+         << "Circle: " << alignof(Circle) << '\n';
+         //When alignas set to 2 then warning appears "alignment cannot be set to less than the default alignment"
+         //and result in 8-alignment
+
     cout << "Areas before sort: " << std::endl;
     printAreas(shapes);
 
+    auto sortByArea = [](shared_ptr<Shape> first, shared_ptr<Shape> second) {
+        if (first == nullptr || second == nullptr)
+            return false;
```
Comment (WBigaj) [inline shapes/main.cpp:98]:
```suggestion
        if (first == nullptr || second == nullptr) {
            return false;
        }
```

File: shapes/main.cpp [hunk 2]
```diff
@@ -103,8 +106,48 @@ int main()
     auto square = make_shared<Square>(4.0);
     shapes.push_back(square);
 
+    auto perimeterBiggerThan20 = [](shared_ptr<Shape> s) {
+        if (s)
+            return (s->getPerimeter() > 20);
```
Comment (WBigaj) [inline shapes/main.cpp:111]:
```suggestion
        if (s) {
            return (s->getPerimeter() > 20);
        }
```

File: shapes/main.cpp [hunk 3]
```diff
@@ -103,8 +106,48 @@ int main()
...
+        return false;
+    };
+
+    auto areaLessThanX = [x = 10](shared_ptr<Shape> s) {
+        if (s)
+            return (s->getArea() < x);
```
Comment (WBigaj) [inline shapes/main.cpp:117]:
```suggestion
        if (s) {
            return (s->getArea() < x);
        }
```

**Review by WBigaj — COMMENTED** [review]
I have checked your code and left my suggestions.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
15 XP za Nowoczesny C++ SFINAE (4447)
5 XP za Nowoczesny C++ alignas (4448)
5 XP za Nowoczesny C++ delegating constructors (4449)
5 XP za Nowoczesny C++ lambda (4450)
5 XP za Nowoczesny C++ structured bindings (4451)
15 XP za Nowoczesny C++ variadic templates (4452)

{ "requirement_ids": [4447, 4448, 4449, 4450, 4451, 4452] }

**Pull request closed without merging**